presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestWriter.java

```diff
@@ -375,13 +379,51 @@ public void testCollectColumnStatisticsOnInsert()
                         "('c_double', null, 0E0, 0E0, null, null, null, null), " +
                         "('c_timestamp', null, 0E0, 0E0, null, null, null, null), " +
                         "('c_varchar', 0E0, 0E0, 0E0, null, null, null, null), " +
+                        "('c_varbinary', null, 0E0, 0E0, null, null, null, null), " +
                         "('c_array', null, 0E0, 0E0, null, null, null, null), " +
                         "('p_varchar', 0E0, 0E0, 0E0, null, null, null, null), " +
-                        "(null, null, null, null, 0E0, null, null, null)) AS x (c_boolean, c_bigint, c_double, c_timestamp, c_varchar, c_array, p_varchar, p_varchar)");
+                        "(null, null, null, null, 0E0, null, null, null)) AS x (column_name, data_size, distinct_values_count, nulls_fraction, row_count, low_value, high_value, histogram)");
 
         dropTableIfExists(tmpTableName);
     }
 
+    @Test
+    public void testColumnFilter()
+    {
+        Session session = buildSessionForTableWrite();
+        String tmpTableName = generateRandomTableName();
+        assertUpdate(session, format("" +
+                "CREATE TABLE %s " +
+                "AS " +
+                "SELECT c_boolean, c_bigint, c_double, c_timestamp, c_varchar, c_varbinary " +
+                "FROM ( " +
+                "  VALUES " +
+                "    (null, null, null, null, null, null), " +
+                "    (true, BIGINT '1', DOUBLE '2.2', TIMESTAMP '2012-08-08 01:00', CAST('abc1' AS VARCHAR), to_ieee754_64(1))," +
+                "    (false, BIGINT '0', DOUBLE '1.2', TIMESTAMP '2012-08-08 00:00', CAST('abc2' AS VARCHAR), to_ieee754_64(2))," +
+                "    (true, BIGINT '2', DOUBLE '3.3', TIMESTAMP '2012-09-09 01:00', CAST('cba1' AS VARCHAR), to_ieee754_64(3)), " +
+                "    (false, BIGINT '1', DOUBLE '2.3', TIMESTAMP '2012-09-09 00:00', CAST('cba2' AS VARCHAR), to_ieee754_64(4)) " +
+                ") AS x (c_boolean, c_bigint, c_double, c_timestamp, c_varchar, c_varbinary)", tmpTableName), 5);
+
+        // BOOLEAN column filter
+        assertQuery(format("SELECT * FROM %s WHERE c_boolean", tmpTableName), "VALUES (true, BIGINT '1', DOUBLE '2.2', TIMESTAMP '2012-08-08 01:00', CAST('abc1' AS VARCHAR), to_ieee754_64(1)), (true, BIGINT '2', DOUBLE '3.3', TIMESTAMP '2012-09-09 01:00', CAST('cba1' AS VARCHAR), to_ieee754_64(3))");
```

Comment:
Did you try only the assertQuery (validating both java and native with the same SQL) without providing the values list ?

Reply:
Fixed.